services/api/service.go

```diff
@@ -1139,7 +1139,7 @@ func (api *RelayAPI) handleGetPayload(w http.ResponseWriter, req *http.Request)
 	log = log.WithField("timestampBeforePublishing", timeBeforePublish)
 	signedBeaconBlock := common.SignedBlindedBeaconBlockToBeaconBlock(payload, getPayloadResp)
 	code, err := api.beaconClient.PublishBlock(signedBeaconBlock) // errors are logged inside
-	if err != nil {
+	if err != nil || code != 200 {
```

Comment:
Minor nit, I think we should try to use the HTTP status fields:
```suggestion
	if err != nil || code != http.StatusOK {
```